src/mps/sweeps.jl

```diff
@@ -60,6 +63,20 @@ function cutoff!(sw::Sweeps,cutoffs::Float64...)::Nothing
   end
 end
 
+function noise!(sw::Sweeps,noises::Float64...)::Nothing
+  Nm = length(noises)
+  N = min(nsweep(sw),Nm)
+  for i=1:N
+    sw.noise[i] = noises[i]
+  end
+  for i=Nm+1:nsweep(sw)
+    # this is different behavior from cutoff!, min/maxdim!, etc.,
+    # but I (CDW) think it's the right thing:
+    # if the noise isn't specified, it should be 0
```

Comment:
Good to think about this. However, first, if the user wants to set the noise on all remaining sweeps to zero, they can just append an explicit 0.0 to the argument list. Second, this is the behavior in the C++ version so it's also good to be consistent with that unless there's a strong reason otherwise. Finally, having the same behavior as the other functions like `maxdim!` is less surprising to the user. So I will change this to act like the others before merging.